### PR TITLE
Add Ansi::RenderHTML command

### LINKED ISSUE #37: [Page: Editor] ANSI colors work for version 1 test runners, but not version 2
This is an example output of the Prolog test runner (version 1):
Screen Shot 2021-05-22 at 14 18 23

It has colors.

After turning colors in Elixir's test runner (version 2), the output looks like this:

Screen Shot 2021-05-23 at 08 30 50

I would expect both `message` (seen on the screenshot) and `output` to support ANSI colors.

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes https://github.com/exercism/v3-beta/issues/37

image

### CODE REVIEW

File: app/javascript/components/editor/TestSummary.tsx [hunk 1]
```diff
@@ -44,19 +47,19 @@ export function TestSummary({
         {isPresent(test.testCode) ? (
           <div className="--info">
             <h3>Code Run</h3>
-            <pre>{test.testCode}</pre>
+            <pre dangerouslySetInnerHTML={{ __html: test.testCodeHtml }}></pre>
```
Comment (ErikSchierboom) [inline app/javascript/components/editor/TestSummary.tsx]: I've also rendered the test code as HTML, as I could imagine that using syntax highlighting colors there might be really beneficial for tracks that can do that.

File: app/javascript/components/editor/TestSummary.tsx [hunk 2]
```diff
@@ -44,19 +47,19 @@ export function TestSummary({
...
             <h3>{messageLabels[test.status]}</h3>
-            <pre>{test.message}</pre>
+            <pre dangerouslySetInnerHTML={{ __html: test.messageHtml }}></pre>
           </div>
         ) : null}
         {isPresent(test.output) ? (
           <div className="--info">
             <h3>Your Output</h3>
-            <pre>{test.output}</pre>
+            <pre dangerouslySetInnerHTML={{ __html: test.outputHtml }}></pre>
```
Comment (ErikSchierboom) [inline app/javascript/components/editor/TestSummary.tsx:69]: I've kept the output as HTML, as student could theoretically use colors in their output, which the test runner could capture.

File: app/serializers/serialize_submission_test_run.rb [hunk 3]
```diff
@@ -31,10 +31,7 @@ def status
   end
 
   def message_html
-    return nil unless message
-
-    sanitized_message = message.gsub("\e\[K", '')
-    Ansi::To::Html.new(sanitized_message).to_html
+    Ansi::RenderHTML.(message)
```
Comment (iHiD) [inline app/serializers/serialize_submission_test_run.rb]: This probably doesn't need a method now.

File: app/serializers/serialize_submission_test_run.rb [hunk 4]
```diff
@@ -45,12 +42,7 @@ def message
   end
 
   def output_html
-    return if test_run.output.blank?
-
-    # The ansi-to-html library does not support unicode escape sequence
-    # See https://github.com/rburns/ansi-to-html/issues/48
-    sanitized_output = test_run.output.gsub("\e\[K", '')
-    Ansi::To::Html.new(sanitized_output).to_html
+    ::Ansi::RenderHTML.(test_run.output)
```
Comment (iHiD) [inline app/serializers/serialize_submission_test_run.rb]: This probably doesn't need a method now (you can just inline it)

**Comment by iHiD** [comment]
Also still red :)

**Comment by iHiD** [comment]
That's all much nicer! Thanks 🙂

**Pull request merged**